File: String/1528. Shuffle String.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
 var restoreString = function(s, indices) {
    let result = s.split('');
    for(let i = 0; i<s.length; i++){
        result[indices[i]]=s[i];
    }
    return result.join('');
    
};
// Runtime: 156 ms, faster than 5.09% of JavaScript online submissions for Shuffle String.
// Memory Usage: 44.9 MB, less than 17.91% of JavaScript online submissions for Shuffle String.

File: String/1528. Shuffle String.cpp
class Solution {
public:
    string restoreString(string s, vector<int>& indices) {
        string new_s =s;
        for(int i=0;i<s.size();i++)
        {
            new_s[indices[i]]=s[i];
        }
        return new_s;
        
    }
};
// Runtime: 9 ms, faster than 72.63% of C++ online submissions for Shuffle String.
// Memory Usage: 15.1 MB, less than 77.64% of C++ online submissions for Shuffle String.
